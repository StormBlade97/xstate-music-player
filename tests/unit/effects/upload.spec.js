import getFile from "../../utils/getMockAudio";
import audioSvc from "@/services/audio";

describe("load and extract audio data actions", () => {
  window.URL.createObjectURL = jest.fn();
  const files = getFile();

  it("child process to load and enrich tracks", async () => {
    const cb = jest.fn();

    await audioSvc(files[0], "the-file-name")(cb);
    expect(cb.mock.calls).toMatchSnapshot();
  });
});
