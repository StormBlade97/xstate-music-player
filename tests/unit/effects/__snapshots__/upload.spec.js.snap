// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`load and extract audio data actions child process to load and enrich tracks 1`] = `
Array [
  Array [
    Object {
      "payload": Object {
        "albumArt": "",
        "artist": undefined,
        "audioElem": <audio
          preload="auto"
        />,
        "duration": "0:00",
        "id": "the-file-name",
        "metadata": Object {
          "common": Object {
            "disk": Object {
              "no": null,
              "of": null,
            },
            "track": Object {
              "no": null,
              "of": null,
            },
          },
          "format": Object {
            "lossless": false,
            "tagTypes": Array [],
          },
          "native": undefined,
          "quality": Object {
            "warnings": Array [],
          },
        },
        "src": undefined,
        "title": undefined,
      },
      "type": "CHILD_UPDATE",
    },
  ],
  Array [
    Object {
      "id": "the-file-name",
      "type": "LOAD_TRACK_SUCCESS",
    },
  ],
]
`;
